File: MultiLanguageSandbox/src/test/deps/cpp/23.cpp
#include <vector>
#include <algorithm>
#include <cassert>
#include <cmath>
#include <cstdio>
#include <cstdlib>
#include <cstring>
#include <cstdarg>
/**
Count the number of acute-angled triangles that can be formed by selecting any 3 edges out of
a given set of edges.
  
Args:
- edges_count: The number of edges passed.
- An arbitrary number of integer arguments representing the lengths of the edges.
  
Returns:
int: The count of distinct acute-angled triangles that can be formed.
  
Examples:
>>> count_acute_triangles(4, 1, 1, 1, 1)
4
**/
using namespace std;

int count_acute_triangles(int edges_count, ...) {
    va_list args;
    va_start(args, edges_count);
    
    vector<int> edges;
    for (int i = 0; i < edges_count; ++i) {
        edges.push_back(va_arg(args, int));
    }
    va_end(args);
    
    int count = 0;
    
    // Iterate through all possible triplets
    for (int i = 0; i < edges_count; ++i) {
        for (int j = i + 1; j < edges_count; ++j) {
            for (int k = j + 1; k < edges_count; ++k) {
                int a = edges[i];
                int b = edges[j];
                int c = edges[k];
                
                // Check triangle inequality
                if (a + b > c && a + c > b && b + c > a) {
                    // Find the longest side
                    int max_side = max({a, b, c});
                    int sum_of_squares = a*a + b*b + c*c;
                    int max_square = max_side * max_side;
                    
                    // Check if it's acute: sum of squares of all sides > 2 * max_square
                    // Or equivalently, sum of squares of the two smaller sides > max_square
                    if (sum_of_squares > 2 * max_square) {
                        count++;
                    }
                }
            }
        }
    }
    
    return count;
}
int main() {
    assert(count_acute_triangles(4, 1, 1, 1, 1) == 4);
    assert(count_acute_triangles(3, 1, 2, 3) == 0);
    // Additional tests to ensure correctness
    assert(count_acute_triangles(5, 3, 4, 5, 7, 10) == 0);
    assert(count_acute_triangles(6, 6, 8, 10, 5, 5, 5) == 4);
    return 0;
}

File: MultiLanguageSandbox/src/test/deps/lua/23.lua

--[[
Count the number of acute-angled triangles that can be formed by selecting any 3 edges out of a given set of edges.

Args:
- edges: A table containing integer values representing the lengths of the edges.

Returns:
int: The count of distinct acute-angled triangles that can be formed.

Examples:
>>> count_acute_triangles({1, 1, 1, 1})
4
]]

function count_acute_triangles(edges)
    local count = 0
    local n = #edges
    -- Sort the edges to facilitate processing
    table.sort(edges)
    
    -- Iterate through all possible triplets
    for i = 1, n - 2 do
        for j = i + 1, n - 1 do
            for k = j + 1, n do
                local a, b, c = edges[i], edges[j], edges[k]
                -- Check triangle inequality: a + b > c
                if a + b > c then
                    -- Check for acute triangle: a^2 + b^2 > c^2
                    if a * a + b * b > c * c then
                        count = count + 1
                    end
                end
            end
        end
    end
    
    return count
end
function testCountAcuteTriangles()
    assert(count_acute_triangles({1, 1, 1, 1}) == 4)
    assert(count_acute_triangles({1, 2, 3}) == 0)
    -- Additional tests to ensure correctness
    assert(count_acute_triangles({3, 4, 5, 7, 10}) == 0)
    assert(count_acute_triangles({6, 8, 10, 5, 5, 5}) == 4)
end

testCountAcuteTriangles()